lib/src/database/hive_collections_database.dart

```diff
@@ -1613,6 +1613,78 @@ class HiveCollectionsDatabase extends DatabaseApi {
       String userId, CachedProfileInformation profile) async {
     return;
   }
+
+  @override
+  Future<void> cancelSend(Event event) async {
+    if (event.status.isSent) {
+      throw Exception('Can only delete events which are not sent yet!');
+    }
+    final room = event.room;
+    final redactedBecause = Event(
+      content: {
+        'string': 'Cancelled send',
+        'redacts': event.eventId,
+      },
+      type: EventTypes.Redaction,
+      eventId: '${event.eventId}_lastItem_local_redaction',
+      senderId: event.senderId,
+      originServerTs: event.originServerTs,
+      room: room,
+    );
+
+    Future<void> updateRoomBox() async {
+      if (room.lastEvent == null) {
+        throw Exception(
+            'tried to set room last event to null, while this is possbile it will move your room to the bottom');
+      }
+      final currentRawRoom = await _roomsBox.get(room.id);
+      if (currentRawRoom != null) {
+        final currentRoom = Room.fromJson(copyMap(currentRawRoom), room.client);
+        currentRoom.lastEvent = room.lastEvent;
+        await _roomsBox.put(
+          room.id,
+          currentRoom.toJson(),
+        );
+      }
+    }
+
+    // redact and store the event itself.
+    event.setRedactionEvent(redactedBecause);
+    await _eventsBox.put(
+      TupleKey(room.id, event.eventId).toString(),
+      event.toJson(),
+    );
+
+    if (room.lastEvent != null && room.lastEvent!.eventId == event.eventId) {
+      Logs().v('cancel send called on lastEvent, setting to redacted');
+
+      // now redact the last event
+      room.lastEvent!.setRedactionEvent(redactedBecause);
+      // and then store it in the db
+      await updateRoomBox();
+
+      // check if room was postLoaded
+      if (!room.partial) {
+        final newLastEvent =
+            (await room.getTimeline()).events.skip(1).firstWhereOrNull(
+                  (event) =>
+                      room.client.roomPreviewLastEvents.contains(event.type),
```

Comment:
You can't just load the last message like this because you don't know if in the timeline even is a potential last event candidate. There could be 100 member/call state events in between. That's why I said we should do it like whatsapp

Reply:
> No the room is already postLoaded, so you probably have a visible event already. And incase the last 30 messages (the default output of getTImeline) does not have a proper last event, it then auotmatically fallsback to what whatsapp does and shows you "redacted"

As we decided this was too confusing I have removed it and kept it at https://github.com/famedly/matrix-dart-sdk/tree/td/intellgientfallbackstash for future viewing.